module/apmecho/middleware.go

```diff
@@ -72,8 +72,9 @@ func (m *middleware) handle(c echo.Context) error {
 	name := req.Method + " " + c.Path()
 	tx, req := apmhttp.StartTransaction(m.tracer, name, req)
 	defer tx.End()
-	c.SetRequest(req)
 	body := m.tracer.CaptureHTTPRequestBody(req)
+	req.WithContext(apm.ContextWithBodyCapturer(req.Context(), body))
```

Comment:
```suggestion
	req = apmhttp.RequestWithContext(req, apm.ContextWithBodyCapturer(req.Context(), body))
```

`req.WithContext` returns a copy with the context set. Without reassigning req, that WithContext is a no-op.
apmhttp.RequestWithContext is a slightly more optimised version of req.WithContext.